workflows/plots.py

```diff
@@ -237,69 +237,92 @@ def plot_sfs(s, outfile):
+    #grapes_bestfits = _read_bestfits(input.grapes_res, ['pop0', 'pop1', 'pop2'])
+
+    dadi_shapes = [dadi_bestfits[i]['shape'] for i in range(len(dadi_bestfits))]
+    dadi_Es = [dadi_bestfits[i]['shape']*dadi_bestfits[i]['scale']/(dadi_bestfits[i]['theta']/(2*mu*seq_len*nonneu_prop)) for i in range(len(dadi_bestfits))]
 
     ## theta_bar = 4*Ne*mu
     ## S_d = 4*Ne*E(s)
     ## E(s) = mu*S_d/theta_bar
     ## See Fig. 1 in doi: 10.1534/genetics.117.300323
-    polydfe_pop0_Es = abs(mu*polydfe_pop0['S_d']/polydfe_pop0['theta_bar'])
-    polydfe_pop1_Es = abs(mu*polydfe_pop1['S_d']/polydfe_pop1['theta_bar'])
-    polydfe_pop2_Es = abs(mu*polydfe_pop2['S_d']/polydfe_pop2['theta_bar'])
+    polydfe_shapes = [polydfe_bestfits[i]['b'] for i in range(len(polydfe_bestfits))]
+    polydfe_Es = [abs(mu*polydfe_bestfits[i]['S_d']/polydfe_bestfits[i]['theta_bar']) for i in range(len(polydfe_bestfits))]
+
+    dfe_alpha_shapes = [dfe_alpha_bestfits[i]['b'] for i in range(len(dfe_alpha_bestfits))]
+    dfe_alpha_Es = [dfe_alpha_bestfits[i]['Es'] for i in range(len(dfe_alpha_bestfits))]
 
     fig = plt.figure(figsize=(6,6), dpi=300)
 
-    #plt.subplot(2,2,1)
-    #plt.ylabel('Shape parameter')
-    #plt.xlabel('Population')
-    #plt.boxplot([dadi_pop0['shape'], dadi_pop1['shape'], dadi_pop2['shape']])
-    #plt.plot(plt.xlim(), [0.19,0.19], c='red')
-    #plt.xticks([1,2,3],['YRI','CEU','CHB'])
-    #plt.ylim([0.1,0.25])
-    #plt.title('dadi')
-
-    #plt.subplot(2,2,2)
-    #plt.ylabel('|E(s)|')
-    #plt.xlabel('Population')
-    #plt.boxplot([dadi_pop0_Es, dadi_pop1_Es, dadi_pop2_Es])
-    #plt.plot(plt.xlim(), [0.014,0.014], c='red')
-    #plt.xticks([1,2,3],['YRI','CEU','CHB'])
-    #plt.ylim([0,0.04])
-    #plt.title('dadi')
-
-    plt.subplot(2,2,3)
-    plt.ylabel('Shape parameter')
-    plt.xlabel('Population')
-    plt.boxplot([polydfe_pop0['b'], polydfe_pop1['b'], polydfe_pop2['b']])
-    plt.plot(plt.xlim(), [0.19,0.19], c='red')
-    plt.xticks([1,2,3],['YRI','CEU','CHB'])
-    plt.ylim([0.1,0.25])
-    plt.title('polyDFE')
+    plt.subplot(3,2,1)
+    _plot_three_pops_boxplots(plt, 'dadi', dadi_shapes, 0.19, 'Shape parameter', [0.1,0.25])
+    plt.subplot(3,2,2)
+    _plot_three_pops_boxplots(plt, 'dadi', dadi_Es, 0.014, '|E(s)|')
 
-    plt.subplot(2,2,4)
-    plt.ylabel('|E(s)|')
-    plt.xlabel('Population')
-    plt.boxplot([polydfe_pop0_Es, polydfe_pop1_Es, polydfe_pop2_Es])
-    plt.plot(plt.xlim(), [0.014,0.014], c='red')
-    plt.xticks([1,2,3],['YRI','CEU','CHB'])
-    #plt.ylim([0,0.04])
-    plt.title('polyDFE')
+    plt.subplot(3,2,3)
+    _plot_three_pops_boxplots(plt, 'polyDFE', polydfe_shapes, 0.19, 'Shape parameter', [0.1,0.25])
+    plt.subplot(3,2,4)
+    _plot_three_pops_boxplots(plt, 'polyDFE', polydfe_Es, 0.014, '|E(s)|')
+
+    plt.subplot(3,2,5)
+    _plot_three_pops_boxplots(plt, 'DFE-alpha', dfe_alpha_shapes, 0.19, 'Shape parameter')
```

Comment:
will this work for single population demographic models?

Reply:
Yes, I changed `_plot_three_pops_boxplots` to `_plot_boxplots`, now it can make plots with a single population